Standard/src/Canon/CommonGates.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Quantum.Simulation.Core;
+
+namespace Microsoft.Quantum.Canon
+{
+    public partial class CX
+    {
+        public override RuntimeMetadata? GetRuntimeMetadata(IApplyData args)
+        {
+            Debug.Assert(args.Value is ValueTuple<Qubit, Qubit>, $"Failed to retrieve runtime metadata for {this.ToString()}.");
+
+            if (args.Value is ValueTuple<Qubit, Qubit> cxArgs)
+            {
+                var (ctrl, target) = cxArgs;
+                return new RuntimeMetadata()
+                {
+                    Label = "X",
+                    IsControlled = true,
+                    Controls = new List<Qubit>() { ctrl },
+                    Targets = new List<Qubit>() { target },
+                };
+            }
+
+            return null;
```

Comment:
You can use C# 8 pattern matching to make this sort of logic a lot more compact.
```suggestion
            return args.Value switch
            {
                (Qubit ctrl, Qubit target) => new RuntimeMetadata()
                {
                    Label = "X",
                    IsControlled = true,
                    Controls = new List<Qubit>() { ctrl },
                    Targets = new List<Qubit>() { target },
                },
                _ => null
            };
```
Heck, you can even collapse further into an expression-body method by moving `Debug.Assert` call into the `_ => null` arm as a throw.

Reply:
Wow, that is actually really nice. I wish I'd known that sooner as there are multiple places where I have this same structure. Love these C# pearls @cgranade!